From 0bdd6972a9d3d526ff3de563e2d82fde0df05825 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Wed, 3 Apr 2019 13:01:43 -0700
Subject: [PATCH 0594/1852] v4l: xilinx: scd: Reorder structure definitions

commit 8cb4b52d9559db34e7d2aa597447824a2f29b425 from
https://github.com/Xilinx/linux-xlnx.git

This will make it easier to refactor them.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../platform/xilinx/xilinx-scenechange.h      | 80 +++++++++----------
 1 file changed, 40 insertions(+), 40 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-scenechange.h b/drivers/media/platform/xilinx/xilinx-scenechange.h
index aee1c4af85d0..cd0cebce50e6 100644
--- a/drivers/media/platform/xilinx/xilinx-scenechange.h
+++ b/drivers/media/platform/xilinx/xilinx-scenechange.h
@@ -66,46 +66,6 @@
 
 /****************************** PROTOTYPES ************************************/
 
-/**
- * struct xscd_shared_data - Data to be shared among v4l subdev and DMA engine
- * @iomem: device I/O register space remapped to kernel virtual memory
- * @dma_chan_list: List of DMA channels available
- * @active_streams: Number of active streams
- * @memory_based: Flag to identify memory based mode
- */
-struct xscd_shared_data {
-	void __iomem *iomem;
-	struct xscd_dma_chan *dma_chan_list[XSCD_MAX_CHANNELS];
-	u8 active_streams;
-	bool memory_based;
-};
-
-/**
- * struct xscd_device - Xilinx Scene Change Detection device structure
- * @iomem: device I/O register space remapped to kernel virtual memory
- * @numstreams: Number of streams in the design
- * @irq: Device IRQ
- * @dev: (OF) device
- * @rst_gpio: reset GPIO
- * @clk: video core clock
- * @dma_device: DMA device pointer
- * @shared_data: Data Shared across devices
- * @dma_node: DMA device node
- * @subdevs: subdev device instance
- */
-struct xscd_device {
-	void __iomem *iomem;
-	int numstreams;
-	int irq;
-	struct device *dev;
-	struct gpio_desc *rst_gpio;
-	struct clk *clk;
-	struct platform_device *dma_device;
-	struct xscd_shared_data shared_data;
-	struct device_node *dma_node;
-	struct platform_device *subdevs[XSCD_MAX_CHANNELS];
-};
-
 /**
  * struct xscd_dma_desc - DMA channel
  * @luma_plane_addr: Luma plane buffer address
@@ -232,6 +192,46 @@ struct xscd_dma_device {
 	u8 memory_based;
 };
 
+/**
+ * struct xscd_shared_data - Data to be shared among v4l subdev and DMA engine
+ * @iomem: device I/O register space remapped to kernel virtual memory
+ * @dma_chan_list: List of DMA channels available
+ * @active_streams: Number of active streams
+ * @memory_based: Flag to identify memory based mode
+ */
+struct xscd_shared_data {
+	void __iomem *iomem;
+	struct xscd_dma_chan *dma_chan_list[XSCD_MAX_CHANNELS];
+	u8 active_streams;
+	u8 memory_based;
+};
+
+/**
+ * struct xscd_device - Xilinx Scene Change Detection device structure
+ * @iomem: device I/O register space remapped to kernel virtual memory
+ * @numstreams: Number of streams in the design
+ * @irq: Device IRQ
+ * @dev: (OF) device
+ * @rst_gpio: reset GPIO
+ * @clk: video core clock
+ * @dma_device: DMA device pointer
+ * @shared_data: Data Shared across devices
+ * @dma_node: DMA device node
+ * @subdevs: subdev device instance
+ */
+struct xscd_device {
+	void __iomem *iomem;
+	int numstreams;
+	int irq;
+	struct device *dev;
+	struct gpio_desc *rst_gpio;
+	struct clk *clk;
+	struct platform_device *dma_device;
+	struct xscd_shared_data shared_data;
+	struct device_node *dma_node;
+	struct platform_device *subdevs[XSCD_MAX_CHANNELS];
+};
+
 /*
  * Register related operations
  */
-- 
2.31.1

